package.json

```diff
@@ -11,47 +11,49 @@
     "typescript"
   ],
   "scripts": {
-    "build": "babel gatsby-* -d . --extensions .ts,.tsx",
-    "prepublishOnly": "yarn build"
+    "tsc": "tsc --emitDeclarationOnly --skipLibCheck",
+    "build": "babel src -d . --extensions .ts,.tsx",
+    "prepublishOnly": "yarn build && yarn tsc"
   },
-  "files": [
-    "index.js",
-    "gatsby-node.js",
-    "types.ts",
-    "images"
-  ],
   "peerDependencies": {
     "gatsby": ">=2.0.0",
-    "graphql": "^0.13.0 || ^14.0.0"
+    "graphql": "^0.13.0 || ^14.0.0",
+    "react": "^16.4.2"
   },
   "dependencies": {
     "@graphql-codegen/core": "^1.12.1",
-    "@graphql-codegen/typescript": "^1.12.1",
-    "@graphql-codegen/typescript-operations": "^1.12.1",
-    "@graphql-codegen/typescript-resolvers": "^1.12.1",
+    "@graphql-codegen/flow": "^1.12.2",
+    "@graphql-codegen/flow-operations": "^1.12.2",
+    "@graphql-codegen/typescript": "^1.12.2",
+    "@graphql-codegen/typescript-operations": "^1.12.2",
     "@graphql-toolkit/code-file-loader": "^0.9.7",
     "@graphql-toolkit/common": "^0.9.7",
     "@graphql-toolkit/core": "^0.9.7",
     "@graphql-toolkit/json-file-loader": "^0.9.7",
-    "@types/normalize-path": "^3.0.0",
+    "async": "^3.1.1",
     "chokidar": "^3.0.2",
-    "lodash.debounce": "^4.0.8",
+    "has-yarn": "^2.1.0",
     "normalize-path": "^3.0.0"
   },
+  "optionalDependencies": {
```

Comment:
:thinking: I'm still not sure this actually make sense

Reply:
I think you'd still need to include these in `dependencies`. AFAIK `optionalDependencies` implies that these dependencies is not required _at all_, and the library will handle that case internally. For us, if a user set `includeResolvers: true` and the `@graphql-codegen/typescript-resolvers` happened to be missing from node_module, the lib will fail.

Reply:
Ah I've just seen the handling code below — I guess that make sense, but IMHO it also make the code unnecessary complex :P